src/components/interactive-map.js

```diff
@@ -27,6 +29,12 @@ export default class InteractiveMap extends PureComponent {
     autobind(this);
   }
 
+  getChildContext() {
```

Comment:
If you render the markers as children of the StaticMap then StaticMap will provide the context. Because we support hiding the map in this component, the markers are not nested under the map.
For the custom control scenario, we could add an optional prop "MapControl" that allows users to pass in a custom component without implementing InteractiveMap again.